cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowHttpHandler.java

```diff
@@ -105,23 +110,25 @@
     .registerTypeAdapter(ScheduleSpecification.class, new ScheduleSpecificationCodec())
     .registerTypeAdapter(WorkflowTokenDetail.class, new WorkflowTokenDetailCodec())
     .registerTypeAdapter(WorkflowTokenNodeDetail.class, new WorkflowTokenNodeDetailCodec())
+    .registerTypeAdapter(Trigger.class, new ProtoTriggerCodec())
+    .registerTypeAdapter(Constraint.class, new ProtoConstraintCodec())
     .create();
 
   private final WorkflowClient workflowClient;
   private final DatasetFramework datasetFramework;
-  private final Scheduler scheduler;
+  private final co.cask.cdap.internal.app.runtime.schedule.Scheduler timeScheduler;
 
   @Inject
   WorkflowHttpHandler(Store store, WorkflowClient workflowClient, ProgramRuntimeService runtimeService,
-                      QueueAdmin queueAdmin, Scheduler scheduler, PreferencesStore preferencesStore,
+                      QueueAdmin queueAdmin, co.cask.cdap.internal.app.runtime.schedule.Scheduler timeScheduler,
                       MRJobInfoFetcher mrJobInfoFetcher, ProgramLifecycleService lifecycleService,
-                      MetricStore metricStore, NamespaceQueryAdmin namespaceQueryAdmin,
+                      MetricStore metricStore, NamespaceQueryAdmin namespaceQueryAdmin, Scheduler scheduler,
                       DatasetFramework datasetFramework, DiscoveryServiceClient discoveryServiceClient) {
-    super(store, runtimeService, discoveryServiceClient, lifecycleService, queueAdmin, preferencesStore,
-          mrJobInfoFetcher, metricStore, namespaceQueryAdmin);
+    super(store, runtimeService, discoveryServiceClient, lifecycleService, queueAdmin,
+          mrJobInfoFetcher, metricStore, namespaceQueryAdmin, scheduler);
     this.workflowClient = workflowClient;
     this.datasetFramework = datasetFramework;
-    this.scheduler = scheduler;
+    this.timeScheduler = timeScheduler;
```

Comment:
I don't think you should name this `timeScheduler`.
Maybe name it `oldScheduler`, or name it `scheduler`, and rename the new one `programScheduler`?